core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -172,8 +172,8 @@ module Streamly.Internal.Data.Fold
     , with
 
     -- ** Transforming the Monad
-    , hoist
-    , generally
+    , morphInner
```

Comment:
Isn't hoist already a common name that developers are familiar with?

Reply:
`hoist` is not intuitive. the word hoist is a synonym for lift, however it does not lift, it transforms the monad. The name does not give any idea about that. I do not know the origin of this usage but it sounds wrong. We have `liftInner` for lifting the inner monad and similarly morphInner for transforming the inner monad makes better sense, the name gives an idea about transformation nature of the operation.